### PR TITLE
feat: Organizations users calendar cache

### PR DESCRIPTION
**Pull request by roae**
## What does this PR do?

Create the ISR calendar-cache page to add support to org users.

## Type of change

- New feature (non-breaking change which adds functionality)

## Mandatory Tasks
- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/core/CalendarManager.ts [hunk 1]
```diff
@@ -185,13 +185,20 @@ export const getCachedResults = async (
...
+  month: string,
+  organizationSlug: string
+): Promise<EventBusyDate[][]> => {
   let localCache: EventBusyDate[][] = [];
   const { NODE_ENV } = process.env;
   const cacheDir = `${NODE_ENV === "development" ? NODE_ENV : process.env.BUILD_ID}`;
   const baseUrl = `${WEBAPP_URL}/_next/data/${cacheDir}/en`;
-  const url = `${baseUrl}/${username}/calendar-cache/${month}.json?user=${username}&month=${month}`;
+  const url = organizationSlug
+    ? `${baseUrl}/${username}/calendar-cache/${month}/{organizationSlug}.json?user=${username}&month=${month}&orgSug=${organizationSlug}`
```
Comment (roae) [inline packages/core/CalendarManager.ts:200]:
```suggestion
    ? `${baseUrl}/${username}/calendar-cache/${month}/{organizationSlug}.json?user=${username}&month=${month}&orgSlug=${organizationSlug}`
```

File: packages/features/ee/organizations/lib/orgDomains.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export function getOrgDomain(hostname: string) {
   const currentHostname = appHostnames.find((ahn) => {
     const url = new URL(WEBAPP_URL);
     const testHostname = `${url.hostname}${url.port ? `:${url.port}` : ""}`;
-    return testHostname.endsWith(`.${ahn}`);
+    return testHostname.endsWith(ahn);
```
Comment (roae) [inline packages/features/ee/organizations/lib/orgDomains.ts:23]: This is unrelated, but I was getting 404 error in organization and users profile.

**Review by leog — APPROVED** [review]
LGTM, nicely done @roae

File: packages/core/CalendarManager.ts [hunk 3]
```diff
@@ -185,13 +185,20 @@ export const getCachedResults = async (
...
+  month: string,
+  organizationSlug: string
+): Promise<EventBusyDate[][]> => {
   let localCache: EventBusyDate[][] = [];
   const { NODE_ENV } = process.env;
   const cacheDir = `${NODE_ENV === "development" ? NODE_ENV : process.env.BUILD_ID}`;
   const baseUrl = `${WEBAPP_URL}/_next/data/${cacheDir}/en`;
-  const url = `${baseUrl}/${username}/calendar-cache/${month}.json?user=${username}&month=${month}`;
+  const url = organizationSlug
+    ? `${baseUrl}/${username}/calendar-cache/${month}/{organizationSlug}.json?user=${username}&month=${month}&orgSlug=${organizationSlug}`
```
Comment (zomars) [inline packages/core/CalendarManager.ts:200]:
```suggestion
    ? `${baseUrl}/${username}/calendar-cache/${month}/${organizationSlug}.json?user=${username}&month=${month}&orgSlug=${organizationSlug}`
```

File: packages/features/auth/lib/getServerSession.ts [hunk 4]
```diff
@@ -46,7 +46,7 @@ export async function getServerSession(options: {
     return cachedSession;
   }
 
-  const user = await prisma.user.findUnique({
+  const user = await prisma.user.findFirst({
```
Comment (zomars) [inline packages/features/auth/lib/getServerSession.ts:49]:
```suggestion
  const user = await prisma.user.findUnique({
```

File: packages/features/auth/lib/next-auth-options.ts [hunk 5]
```diff
@@ -69,7 +69,7 @@ const providers: Provider[] = [
         throw new Error(ErrorCode.InternalServerError);
       }
 
-      const user = await prisma.user.findUnique({
+      const user = await prisma.user.findFirst({
```
Comment (zomars) [inline packages/features/auth/lib/next-auth-options.ts:72]:
```suggestion
      const user = await prisma.user.findUnique({
```

File: packages/features/ee/sso/lib/saml.ts [hunk 6]
```diff
@@ -31,7 +31,7 @@ export const isSAMLAdmin = (email: string) => {
 };
 
 export const samlTenantProduct = async (prisma: PrismaClient, email: string) => {
-  const user = await prisma.user.findUnique({
+  const user = await prisma.user.findFirst({
```
Comment (zomars) [inline packages/features/ee/sso/lib/saml.ts:34]:
```suggestion
  const user = await prisma.user.findUnique({
```

File: packages/lib/constants.ts [hunk 7]
```diff
@@ -8,7 +8,7 @@ export const WEBAPP_URL =
   RAILWAY_STATIC_URL ||
   HEROKU_URL ||
   RENDER_URL ||
-  "http://localhost:3000";
+  "http:/localhost:3000";
```
Comment (zomars) [inline packages/lib/constants.ts:11]:
```suggestion
  "http://localhost:3000";
```

File: packages/trpc/server/routers/loggedInViewer/deleteMe.handler.ts [hunk 8]
```diff
@@ -20,7 +20,7 @@ type DeleteMeOptions = {
 
 export const deleteMeHandler = async ({ ctx, input }: DeleteMeOptions) => {
   // Check if input.password is correct
-  const user = await prisma.user.findUnique({
+  const user = await prisma.user.findFirst({
```
Comment (zomars) [inline packages/trpc/server/routers/loggedInViewer/deleteMe.handler.ts:23]:
```suggestion
  const user = await prisma.user.findUnique({
```

File: packages/trpc/server/routers/loggedInViewer/deleteMeWithoutPassword.handler.ts [hunk 9]
```diff
@@ -12,7 +12,7 @@ type DeleteMeWithoutPasswordOptions = {
 };
 
 export const deleteMeWithoutPasswordHandler = async ({ ctx }: DeleteMeWithoutPasswordOptions) => {
-  const user = await prisma.user.findUnique({
+  const user = await prisma.user.findFirst({
```
Comment (zomars) [inline packages/trpc/server/routers/loggedInViewer/deleteMeWithoutPassword.handler.ts:15]:
```suggestion
  const user = await prisma.user.findUnique({
```

**Review by zomars — APPROVED** [review]

**Pull request merged**